src/main/java/com/google/api/codegen/PageStreamingConfig.java

```diff
@@ -77,20 +83,37 @@ public static PageStreamingConfig createPageStreaming(
               resourcesFieldName));
     }
 
+    // page_size is allowed to be null.
     if (requestTokenField == null || responseTokenField == null || resourcesField == null) {
       return null;
     }
-    return new PageStreamingConfig(requestTokenField, responseTokenField, resourcesField);
+    return new PageStreamingConfig(pageSizeField, requestTokenField, responseTokenField, resourcesField);
   }
 
   private PageStreamingConfig(
-      Field requestTokenField, Field responseTokenField, Field resourcesField) {
+      Field pageSizeField, Field requestTokenField, Field responseTokenField, Field resourcesField) {
+    this.pageSizeField = pageSizeField;
     this.requestTokenField = requestTokenField;
     this.responseTokenField = responseTokenField;
     this.resourcesField = resourcesField;
   }
 
   /**
+   * Returns true if page_size is not null.
+   */
+  public boolean hasPageSizeField() {
+    return pageSizeField != null;
```

Comment:
Note, this is really just so I can check for the field in snippet language... since null's aren't allowed

If someone knows a way around this that doesn't require a new method please let me know.